### PR TITLE
Add marker storage feature which saves markers users put on the map.

### PR DESCRIPTION
**Pull request by josefj1519**
Assignee: LiRq95
Screenshot: doEhyH4eoFp

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    response.getWriter().println(new Gson().toJson(markers));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) {
+    double lat = Double.parseDouble(request.getParameter("lat"));
+    double lng = Double.parseDouble(request.getParameter("lng"));
+
+    Marker marker = new Marker(lat, lng);
+    storeMarker(marker);
```
Comment (LiRq95) [inline portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java:48]: As the variables above are not reused, you can probably refactor lines 44-48 into 1 line without sacrificing readability.

File: portfolio/src/main/java/com/google/sps/data/Marker.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+/** Represents a marker on the map. */
+public class Marker {
```
Comment (LiRq95) [inline portfolio/src/main/java/com/google/sps/data/Marker.java:18]: Not sure if you can access it from your project, but if you'd like, I believe you could use an existing map api's class for latitude/longitude tuples at https://github.com/googlemaps/google-maps-services-java/blob/b0a48afef86b2b6431821ea7d0f4c441674957f0/src/main/java/com/google/maps/model/LatLng.java#L24

It's compatible with some other API code in the same package which can let you do some cool geo stuff if you feel like it, and can give you url values if you ever want to put the marker into a link. Anyways it should save you from writing some impls (if you wanna add more features to the class itself later you could always make a class that inherits from it).

File: portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  private Collection<Marker> getMarkers() {
+    Collection<Marker> markers = new ArrayList<>();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Marker");
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()) {
+      double lat = (double) entity.getProperty("lat");
+      double lng = (double) entity.getProperty("lng");
+      Marker marker = new Marker(lat, lng);
+      markers.add(marker);
```
Comment (LiRq95) [inline portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java:61]: 58-61 can prolly be 1 line w/ same justification as comment above.

Comment (josefj1519) [reply portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java]: Fixed.

File: portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java (hunk 1 above)
Comment (josefj1519) [reply portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java]: Fixed.

File: portfolio/src/main/java/com/google/sps/data/Marker.java (hunk 2 above)
Comment (josefj1519) [reply portfolio/src/main/java/com/google/sps/data/Marker.java]: Done.  This is a good bookmark if we chose to do anything with maps in the capstone.  Right now I'm just going to be using the tuples.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -68,27 +68,52 @@ async function getUserImages(){
...
 
+let map;
+
+let editMarker;
+
 function createMap() {
- // Lattitude and longitude of Chavez Ravine.
-  var chavezRavine = {lat: 34.073851, lng: -118.242147};
-  const map = new google.maps.Map(
+  // Lattitude and longitude of the USA.
```
Comment (LiRq95) [inline portfolio/src/main/webapp/script.js:76]: Fix typo for latitude when you have the chance.

Comment (josefj1519) [reply portfolio/src/main/webapp/script.js]: Fixed.

**Pull request merged**